include/tvm/relay/op_attr_types.h

```diff
@@ -158,13 +158,15 @@ using FTVMAlterOpLayout = runtime::TypedPackedFunc<
  * \param tinfos An array of placeholders, use for getting the inferred shape
  *               and dtype of the inputs.
  * \param desired_layout The desired layout.
+ * \param additional_layouts Specify additional layouts, e.g. kernel_layout.
  * \return new_expr The modified expression.
  */
 using FTVMConvertOpLayout = runtime::TypedPackedFunc<
   Expr(const Attrs& attrs,
        const Array<Expr>& args,
        const Array<te::Tensor>& tinfos,
-       const std::string& desired_layout)>;
+       const std::string& desired_layout,
```

Comment:
Does it make sense to convert desired_layout to a Map from `op_name` to a list of layouts?
For example, it can look like
~~~
"nn.conv2d" : ["NCHW", "IOHW"]
"nn.conv3d" : ["NCDHW", "IODHW"]
~~~

Though I agree that the current interface is very restrictive, additional layout seems little awkward. With the above proposal, as we define convert_layout for the operators, we have a fixed length list that we can use.

Reply:
Thanks, I hope I understood correctly. So the idea is that someone using this pass would specify a map of operator -> [layouts] e.g. `relay.transform.ConvertLayout({"nn.conv2d": ["NCHW", "IOHW"], ...})`? My intention for doing it the additional_layouts way was to minimise the impact on the current state of the pass, although I like your idea.

My only concern with using a list would be an operator that had say 3 different input tensors with different layouts (I don't think one exists currently, could this pop up in the future?). For example data, kernel, some_other. How could we specify a preferred layout for data and some_other whilst leaving kernel set to default?

Reply:
You are correct in understanding my proposal.

Thats interesting. Although, we don't any operator with three different layouts. For this, maybe, we can use a string "default" (or maybe a better name), that leaves the layout for the tensor in that index unchanged. 

Therefore, we can be stricter while calling the ConvertLayout pass. The user must define the layouts for all the tensors. If he/she does not care about a certain tensor layout, even then they must say "default".